build/release/teamcity-publish-release.sh

```diff
@@ -162,3 +162,49 @@ else
   echo "The ${dockerhub_repository}:latest docker image tag was _not_ pushed."
 fi
 tc_end_block "Tag docker image as latest"
+
+
+tc_start_block "Verify docker images"
+
+images=(
+  "${dockerhub_repository}:${build_name}"
+  "${gcr_repository}:${build_name}"
+)
+if [[ -z "$PRE_RELEASE" ]]; then
+  images+=("${dockerhub_repository}:latest-${release_branch}")
+fi
+if [[ -n "${PUBLISH_LATEST}" ]]; then
+  images+=("${dockerhub_repository}:latest")
+fi
+
+errorred=0
+
+for img in "${images[@]}"; do
+  docker rmi "$img"
+  docker pull "$img"
+  output=$(docker run "$img" version)
+  build_type=$(grep "^Build Type:" <<< "$output" | cut -d: -f2 | sed 's/ //g')
+  sha=$(grep "^Build Commit ID:" <<< "$output" | cut -d: -f2 | sed 's/ //g')
+  build_tag=$(grep "^Build Tag:" <<< "$output" | cut -d: -f2 | sed 's/ //g')
```

Comment:
Note: `Build Type`, `Build Commit ID` and `Build Tag` are not all consistent across cockroach release series. Some of them changed in 20.2. If you backport, you'll need to update this for 20.2's names.

Reply:
I just verified and it looks that the supported branches look the same:
```
$ docker run -ti cockroachdb/cockroach:v20.2.12 version
Build Tag:        v20.2.12
Build Time:       2021/06/28 17:42:49
Distribution:     CCL
Platform:         linux amd64 (x86_64-unknown-linux-gnu)
Go Version:       go1.13.14
C Compiler:       gcc 6.3.0
Build Commit ID:  402d4f690a9aebe10042481064415150ab3ead32
Build Type:       release

$ docker run -ti cockroachdb/cockroach:v21.1.4 version
Build Tag:        v21.1.4
Build Time:       2021/06/29 19:06:32
Distribution:     CCL
Platform:         linux amd64 (x86_64-unknown-linux-gnu)
Go Version:       go1.15.11
C Compiler:       gcc 6.5.0
Build Commit ID:  92e342b8f0032a5771d3e50932df4c6d7d85fe41
Build Type:       release
```

It should be safe to backport to 20.2, I guess.

Reply:
Ah, correction on what I stated:

> ... you'll need to update this for 20.2's names.

you'll need to update this for 20.1's names.

And it's `Build Commit ID` that used to be called `Build SHA-1` in 20.1.

```
$ docker run -it cockroachdb/cockroach:v20.1.8 version
Build Tag:    v20.1.8
Build Time:   2020/10/21 15:46:38
Distribution: CCL
Platform:     linux amd64 (x86_64-unknown-linux-gnu)
Go Version:   go1.13.9
C Compiler:   gcc 6.3.0
Build SHA-1:  ffd029f51aa134f2bce4a39ef1f3ad095c3856ad
Build Type:   release
```